File: BOJ/11000/11055_가장큰증가부분수열.cpp
#include <stdio.h>
#include <algorithm>
#define MAX 1001

using namespace std;

int numbers[MAX];
int dp[MAX];

int main(void)
{
    int n;
    scanf("%d", &n);

    for (int i = 0; i < n; i++)
    {
        scanf("%d", &numbers[i]);
        dp[i] = numbers[i];
    }

    for (int i = 1; i < n; i++)
    {
        for (int j = 0; j < i; j++)
        {
            if (numbers[i] > numbers[j])
                dp[i] = max(dp[i], dp[j] + numbers[i]);
        }
    }

    printf("%d", *max_element(dp, dp + n));
}

File: BOJ/11000/11055_가장큰증가부분수열.py
n = int(input())
numbers = list(map(int, input().split()))
dp = list(numbers[i] for i in range(n))

for i in range(1, n):
    for j in range(i):
        if numbers[j] < numbers[i]:
            dp[i] = max(dp[i], dp[j]+numbers[i])

print(max(dp))
